csharp/src/Telemetry/Traces/Exporters/FileExporter/FileExporter.cs

```diff
@@ -120,28 +120,54 @@ public override ExportResult Export(in Batch<Activity> batch)
             foreach (Activity activity in batch)
             {
                 if (activity == null) continue;
-                _activityQueue.Enqueue(activity);
+                _ = _channel.Writer.TryWrite(activity);
             }
             return ExportResult.Success;
         }
 
+        protected override bool OnForceFlush(int timeoutMilliseconds)
+        {
+            CancellationTokenSource cts = new CancellationTokenSource();
+            cts.CancelAfter(TimeSpan.FromMilliseconds(timeoutMilliseconds));
+            Task.Delay(100).ConfigureAwait(false).GetAwaiter().GetResult();
```

Comment:
If these calls are always blocking for the full duration, isn't it much cleaner to just use `Thread.Sleep`?